enviro/plot.py

```diff
@@ -398,74 +393,211 @@ def data_to_table(matrix, var_names):
+        for i in range(2):
+            process = Popen(
+                ['pdflatex', '-output-directory', tempdir],
+                stdin=PIPE,
+                stdout=PIPE,
+            )
+            process.communicate(rendered_tpl)
+        with open(os.path.join(tempdir, 'texput.pdf'), 'rb') as f:
+            pdf = f.read()
+
+
+        short_file_path_report = user + '/latex_report.pdf'
+        full_file_path_report = 'enviro/static/' + short_file_path_report
+        with open(full_file_path_report, 'wb') as f:
+            f.write(pdf)
+
+    return short_file_path_report
+
+def get_latex_eedc_table(matrix, var_names, var_symbols):
     """
-    The function generates a pdf. The pdf includes an image of the contour and a table with the data points.
-    :param matrix:      data points supported by compute. 
-    :param user:        who starts the order. 
-    :param method_label:      e.g. "T = 25 years, IFORM"
-    :param probabilistic_model:       probabilistic model object.
-    :param var_names:   names of the variables.
-    :param var_symbols: symbols of the variables of the probabilistic model
-    :return:            the path to the user related pdf.
+        Creates a latex string containing a table listing the contour's extreme environmental design conditions (EEDCs).
+
+        Parameters
+        ----------
+        matrix : n-dimensional matrix
+            The coordinates of the environmental contour.
+            The format is defined by compute_interface.iform()
+
+        var_names : list of strings
+            Names of the environmental variables used in the probabilistic model,
+            e.g. ['wind speed [m/s]', 'significant wave height [m]']
+
+        var_symbols : list of strings
+            Symbols of the environental variables used in the probabilistic model,
+            e.g. ['V', 'Hs']
+
+        Returns
+        -------
+        table_string : string,
+            A string in latex format containing a table, which lists the first X extreme environmental design conditions
+
+        """
+
+    MAX_EEDCS_TO_LIST_IN_TABLE = 100
+    LINES_FOR_PAGE_BREAK = 40
+
+    reached_max_eedc_number = 0
+
+    table_string = r"\begin{tabular}{"
+    table_string += get_latex_eedc_table_head_line(var_names)
+
+    for i in range(len(matrix[0][1])):
+        table_string += r"" + str(i+1) + r" & "
+        for j in range(len(var_names)):
+            table_string += "{0:.2f}".format(matrix[0][j][i]) # thanks to https://stackoverflow.com/questions/455612/limiting-floats-to-two-decimal-points
+            if j == len(var_names) - 1:
+                table_string += r"\\"
+            else:
+                table_string += r" & "
+        if i % LINES_FOR_PAGE_BREAK == 0 and i > 0:
+            table_string += r"\end{tabular}"
+            table_string += r"\newpage"
+            table_string += r"\begin{tabular}{"
+            table_string += get_latex_eedc_table_head_line(var_names)
```

Comment:
consider storing the title head in a variable instead of generating it repeatedly.